iree/turbine/kernel/wave/runtime/CMakeLists.txt

```diff
@@ -0,0 +1,66 @@
+cmake_minimum_required(VERSION 3.15...3.27)
+project(wave_runtime)
+
+if (CMAKE_VERSION VERSION_LESS 3.18)
+  set(DEV_MODULE Development)
+else()
+  set(DEV_MODULE Development.Module)
+endif()
+
+find_package(Python 3.10 COMPONENTS Interpreter ${DEV_MODULE} REQUIRED)
+
+if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
+
+# Detect the installed nanobind package and import it into CMake
+execute_process(
+  COMMAND "${Python_EXECUTABLE}" -m nanobind --cmake_dir
+  OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE nanobind_ROOT)
+find_package(nanobind CONFIG REQUIRED)
+
+# Link against torch, using the libraries and include dirs in site-packages.
+execute_process(
+  COMMAND "${Python_EXECUTABLE}" -c "import torch; print(torch.utils.cmake_prefix_path)"
+  OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE torch_cmake_PATH)
+list(APPEND CMAKE_PREFIX_PATH ${torch_cmake_PATH})
+list(APPEND CMAKE_PREFIX_PATH "/opt/rocm")
+
+find_package(Torch REQUIRED)
+find_package(HIP REQUIRED)
```

Comment:
These required deps are going to be very painful across systems and platforms. What does the runtime actually need from these packages? Could we use header files and use dynamic linking to access APIs that we need?

Reply:
but nanobind without cmake may be difficult

Reply:
We need to link against torch and hip. For linking against torch, I believe find_package is the most robust way but open to other suggestions? find_package for Torch requires rocm to be installed on the system (this is all of rocm including hipblaslt, hipsparse etc.)(https://github.com/pytorch/pytorch/blob/1b373f6cd4bfe3ffdbab8aeee06decb1dc656f7d/cmake/public/LoadHIP.cmake#L146) but our rocm builders should have this already installed?